### ISSUE TITLE
Google Tag Manager event issue

### ISSUE THREAD

**Issue by kukac7** [angulartics/angulartics2#145]
* **I'm submitting a ...**
[x] bug report
[ ] feature request
[ ] question about the decisions made in the repository

* **Do you want to report a *bug*?**
All `undefined` you send.
I have done everything as described.
https://github.com/angulartics/angulartics2/wiki/Google-Tag-Manager

```javascript
this.angulartics2.eventTrack.next({ action: 'myAction', properties: { category: 'myCategory' }});
```

* **What is the current behavior?**
![screen-shot-2017-09-01-at-16 28 10](https://user-images.githubusercontent.com/1240104/29974459-370284a0-8f33-11e7-89af-0b6b9b2a2cf7.jpg)

* **Please tell us about your environment:**
Angular version: 4.3.5

**Comment by scttcper** [comment]
Its doing what the code is telling it to do.
https://github.com/angulartics/angulartics2/blob/49fc9dee47be803d523185eb3970e64f3c24686f/src/providers/gtm/angulartics2-gtm.ts#L58-L66
Let me know if you think this is unexpected.

**Comment by kukac7** [comment]
@scttcper how can `'interaction'` be overwritten?

```
<button type="button"
angulartics2On="click"
angularticsEvent="eventTracker"
angularticsAction="DownloadClick"
angularticsLabel="label-name"
angularticsValue="value"
angularticsCategory="song"
[angularticsProperties]="{'custom-property': 'Fall Campaign'}">BTN</button>```

```
this.angulartics2.eventTrack.next({
			event: 'eventTracker',
			action: 'Action',
			properties: {
				category: 'Category',
				label: 'Label',
			},
		});```

and your own properties are not sent.

**Comment by scttcper** [comment]
i assume anything passed to gtmCustom will override whatever you're looking for https://github.com/angulartics/angulartics2/blob/master/src/lib/providers/gtm/angulartics2-gtm.ts#L65

**Comment by etonyali** [comment]
Hello,

I am having the same issue. Trying to trigger angulartics2 from code and based on event is not working.

After user signs up successfully, I am trying to run
`this.TM.eventTrack.next({
                        action: 'clicked',
                        properties: {
                            category: 'successfully signed up'
                        },
                    });`

and it is not triggering because there's no event. However when I add the event:
 `this.TM.eventTrack.next({
                        action: 'clicked',
                        event: 'register',
                        properties: {
                            category: 'successfully signed up'
                        }`
it still not triggering.

I tried to put the even in the properties as well, still no luck.
Anybody have any advice?
Basically, I am trying to trigger a tag after user signs up from the code.

**Comment by etonyali** [comment]
If anybody else having this issue, instead of using this package, you can just do:
`(<any>window).dataLayer.push({
                        'category': 'category name',
                        'action': 'action',
                        'event': 'event name'
                    });`

**Comment by tolgabalci** [comment]
I'm having the same issue.  Doing a dataLayer.push() did not fix the issue either.

This is very simple and I followed the setup direction carefully.  My Real-Time overview is correctly showing pageviews, but the below code, inside by button click does not trigger an "Event" under Real-Time => Events or Behavior => Events => Overview.

```
      this.angulartics2.eventTrack.next({ action: 'myAction', properties: { category: 'myCategory' } });
      this.angulartics2.eventTrack.next({ action: 'custom', properties: { category: 'search', label: 'requested' } });
      (<any>window).dataLayer.push({ 'category': 'myCat', 'action': 'myAction', 'event': 'myEvent' });
```
I am using Angular 5.2 with Angulartics2@5.4.0 (the last version that will run on Angular 5)

**Comment by etonyali** [comment]
dataLayer should be working because it's from actual Google Tag Managers code.
More info: https://developers.google.com/tag-manager/devguide#datalayer

Also, this package's google tag manager section was updated 11 days ago. Make sure you look into it and inspect the code.
File: https://github.com/angulartics/angulartics2/blob/master/src/lib/providers/gtm/gtm.ts
GTM Link: https://github.com/angulartics/angulartics2/tree/master/src/lib/providers/gtm

I haven't tested it yet. I'm using angular 6. I will give an update if I do.

**Comment by lucasklaassen** [comment]
I'm wondering if I'm missing something here... not able to get events to push to the dataLayer on Angular 6:
imported into my main module:
```
Angulartics2Module.forRoot([ Angulartics2GoogleTagManager ])
```
My component I want to track events in:
```
constructor(
    private angulartics2: Angulartics2
  ) { }

ngOnInit() {
     this.angulartics2.eventTrack.next({ action: 'start-game', properties: { category: 'casino', label: 'blackjack' }});
}
```

Inspecting the dataLayer after my component has loaded does not include my event. No console errors.

I can push to the dataLayer using this:
```
(<any>window).dataLayer.push({ 'category': 'myCat', 'action': 'myAction', 'event': 'myEvent' });
```

**Comment by lucasklaassen** [comment]
@mjankucic-itk sorry wasn't able to figure it out, I ended up rolling my own DataLayer implementation.

**Comment by revok** [comment]
Guys, I was having the same issue - events simply weren't showing up in my dataLayer. After some snooping (and plenty of cursing) I figured out that it was my uBlock origin plugin that was preventing Google tag manager from working properly. You may want to verify that the problem is not due to some browser extension.

**Comment by pankhuri** [comment]
What is the way to track events using angulartics2 if I have gtag.js snippet in my GA account?

**Comment by Exxenmann** [comment]
Angular 6.0.3

I have no working solution neither via data layer push nor via "this.angulartics2.eventTrack.next({" Both executed and tested correctly.
Data layer push method showed me the data in gtm preview mode but didnt fire any event.
Does anybody have any hint for me?

**Comment by ChBernat** [comment]
@pankhuri Probably move to GTM, define custom events and use them like this. If I will not forget I will update my answer here with results of my work on this theory.

**Comment by chriszrc** [comment]
If people are using GTM, make sure to read this response:

https://github.com/angulartics/angulartics2/issues/99#issuecomment-282915081

Angulartics will not fire any events, even if you do the:

```
this.angulartics2.eventTrack.next( ...etc)
```

unless GTM is setup correctly to respond to events. You need to create a custom tag, with the custom event and custom variables for this to start working-

**Comment by lucasklaassen** [comment]
@chriszrc when I was testing this back in June I was inspecting the dataLayer itself to see if anything was being pushed to it and it wasn't. Has that bug been fixed in a later version?

**Comment by intellix** [comment]
I think event/action are wrong in the GTM adapter. For every other adapter the "action" is basically the abstraction for identifier/event-name.

I've got something like this throughout my codebase:
```ts
angulartics.eventTrack.next({
  action: 'purchaseItem',
  properties: {
    value: 5,
    currency: 'USD'
  },
});
```

When in GTM and I try to create a trigger on the event: `purchaseItem` it never fires because the event name is actually `interaction` if `properties.event` is undefined.

Action in the GTM adapter is passed as an arbitrary property called `action` which you can't really react on.

In my opinion it should follow the same convention and be:
```ts
this.pushLayer({
  event: action || 'interaction',
  target: properties.category || 'Event',
  label: properties.label,
  value: properties,
  interactionType: properties.noninteraction,
  userId: this.angulartics2.settings.gtm.userId,
  ...properties.gtmCustom
});
```

Otherwise you have to do something like:
```ts
angulartics.eventTrack.next({
  action: 'purchaseItem',
  properties: {
    event: 'purchaseItem',
    value: 5,
    currency: 'USD'
  },
});
```
